proto/connectrpc/conformance/v2/client_compat.proto

```diff
@@ -73,6 +74,38 @@ message ClientCompatRequest {
   // For client- or bidi-streaming requests, this delay should be
   // applied before each request sent.
   uint32 request_delay_ms = 18;
+  // If present, the client should cancel the RPC instead of
+  // allowing to complete normally.
+  Cancel cancel = 19;
+  message Cancel {
+    // These fields determine the timing of cancellation.
+    // If none are present, the client should cancel immediately
+    // after all request messages are sent the stream half-closed
+    // (as if the after_half_close_ms field were present and zero).
+    oneof cancel_timing {
+      // When present, the client should cancel instead of
+      // half-closing the stream, after all requests have been
+      // sent. This applies only to client and bidi stream RPCs.
+      google.protobuf.Empty before_half_close = 1;
+      // When present, the client should delay for this many
+      // milliseconds after half-closing the stream and then
+      // cancel.
+      uint32 after_half_close_ms = 2;
+      // When present, the client should cancel right after
+      // reading this number of response messages from the stream.
+      // This applies only to server and bidi stream RPCs.
+      int32 after_num_responses = 3;
+    }
+  }
+
+  // The following field is only used by the reference client. If
+  // you are implementing a client under test, you may ignore it
+  // or respond with an error if the client receives a request where
+  // it is set.
+  //
+  // The raw request may be specified instead of the above group
+  // of fields for defining the request to send.
+  RawHTTPRequest raw_request = 20;
```

Comment:
Should we group this with the first 8 fields above since it also represents a field that shouldn't be explicitly set by tests? Just a thought 🤷‍♂️

Reply:
No, it _should_ be explicitly set by tests. The way you write a test case that uses a raw request is to explicitly set it in the test case _instead of_ the previous group of fields.